index.js

```diff
@@ -26,7 +26,7 @@ function _fillMemDuplex(memDuplexes, index, dataRetrievalFn, response, logger){
         (err, readable) => {
         if(err){
             logger.error('failed to get full object');
-            return response.end();
+            throw new Error('failed to get full object');
```

Comment:
we have think about sending headers in the res.end with @rahulreddy , i will do some test in the week end to have a clean implementation